shared/src/domain-logic/opportunityFields.test.ts

```diff
@@ -0,0 +1,240 @@
+        expect(response).toEqual(false);
+      }
+    );
+
+    it(
+      "returns true when municipality, existing employees, home based business, and ownership type id have been answered and it is  " +
+        "a starting business",
+      async () => {
+        const startingBusiness = generateProfileData({
+          businessPersona: "STARTING",
+          legalStructureId: "limited-partnership",
+          existingEmployees: "1",
+          homeBasedBusiness: true,
+          ownershipTypeIds: ["none"],
+          municipality: generateMunicipality({}),
+        });
+
+        const response = isEveryFundingFieldAnswered(startingBusiness);
+        expect(response).toEqual(true);
+      }
+    );
+
+    it(
+      "returns false when municipality is undefined and it is  " +
+        "a foreign nexus business with a location in NJ",
+      async () => {
+        const startingBusiness = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: true,
+          legalStructureId: "limited-partnership",
+          existingEmployees: "1",
+          homeBasedBusiness: true,
+          ownershipTypeIds: ["none"],
+          municipality: undefined,
+        });
+
+        const response = isEveryFundingFieldAnswered(startingBusiness);
+        expect(response).toEqual(false);
+      }
+    );
+
+    it(
+      "returns true when municipality, existing employees, home based business, and ownership type id have been answered and it is  " +
+        "a foreign nexus business with a location in NJ",
+      async () => {
+        const startingBusiness = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: true,
+          legalStructureId: "limited-partnership",
+          existingEmployees: "1",
+          homeBasedBusiness: true,
+          ownershipTypeIds: ["none"],
+          municipality: generateMunicipality({}),
+        });
+
+        const response = isEveryFundingFieldAnswered(startingBusiness);
+        expect(response).toEqual(true);
+      }
+    );
+
+    it(
+      "returns false when home based business is undefined and it is  " +
+        "a foreign nexus business that is not in NJ",
+      async () => {
+        const startingBusiness = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: false,
+          legalStructureId: "limited-partnership",
+          existingEmployees: "1",
+          homeBasedBusiness: undefined,
+          ownershipTypeIds: ["none"],
+          municipality: generateMunicipality({}),
+        });
+
+        const response = isEveryFundingFieldAnswered(startingBusiness);
+        expect(response).toEqual(false);
+      }
+    );
+
+    it(
+      "returns true when home based business is true and it is  " +
```

Comment:
[pebble] Again, the test name should describe the _behavior_ not the actual test instance (same feedback as the GP/trade name comment)

The _test_ has `homeBasedBusiness` set to true, yes, but the behavior under test is that it returns true if the `homeBasedBusiness` question is _answered_ (ie, either true OR false would be a valid value)

So we should say that - when it has been answered. Or we should test explicitly for the false case as well. Right now the way it's describing the test is leaving a gap in the understanding... we know what happens if it's undefined and if it's true, but what if it's false? The tests don't say. Whereas testing for if undefined and if not undefined does cover every possible behavior.

Reply:
Done